File: backend/app/models/teacher_subject.py
from sqlalchemy import Column, Integer, ForeignKey, UniqueConstraint
from app.database import Base

class TeacherSubject(Base):
    __tablename__ = "teacher_subjects"

    id = Column(Integer, primary_key=True, index=True)
    teacher_id = Column(Integer, ForeignKey("teachers.id", ondelete="CASCADE"))
    subject_id = Column(Integer, ForeignKey("subjects.id", ondelete="CASCADE"))
    class_id = Column(Integer, ForeignKey("classes.id", ondelete="CASCADE"))

    __table_args__ = (
        UniqueConstraint("teacher_id", "subject_id", "class_id", name="uq_teacher_subject_class"),
    )

File: SmartTimetable/src/SmartTimetable.Core/Entities/TeacherSubject.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartTimetable.Core.Entities;

[Table("teacher_subjects")]
public class TeacherSubject
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("teacher_id")]
    public int? TeacherId { get; set; }

    [Column("subject_id")]
    public int? SubjectId { get; set; }

    [Column("class_id")]
    public int? ClassId { get; set; }

    // Navigation
    [ForeignKey("TeacherId")]
    public Teacher? Teacher { get; set; }

    [ForeignKey("SubjectId")]
    public Subject? Subject { get; set; }

    [ForeignKey("ClassId")]
    public Class? Class { get; set; }
}
